src/Symfony/Component/Messenger/Bridge/GooglePubSub/composer.json

```diff
@@ -0,0 +1,40 @@
+{
+    "name": "symfony/google-pubsub-messenger",
+    "type": "symfony-bridge",
+    "description": "Symfony Google Pub/Sub extension Messenger Bridge",
+    "keywords": [],
+    "homepage": "https://symfony.com",
+    "license": "MIT",
+    "authors": [
+        {
+            "name": "Fabien Potencier",
+            "email": "fabien@symfony.com"
+        },
+        {
+            "name": "Symfony Community",
+            "homepage": "https://symfony.com/contributors"
+        }
+    ],
+    "require": {
+        "php": ">=7.2.5",
+        "ext-json": "*",
+        "google/cloud-pubsub": "^1.0",
```

Comment:
This will download the following packages: 

```
google/protobuf
google/common-protos
ralouphie/getallheaders
psr/http-message
guzzlehttp/psr7
guzzlehttp/promises
grpc/grpc 
firebase/php-jwt
psr/http-client
guzzlehttp/guzzle
google/auth 
google/grpc-gcp
google/gax 
monolog/monolog 
rize/uri-template 
google/cloud-core
google/cloud-pubsub
```

All that for sending 4 HTTP request. Could we not depend on an external package like this? Maybe do the HTTP requests with the Symfony HTTP client?

Reply:
You are right. The official SDK surely is the most convenient approach but it comes with Guzzle etc. Using the HttpClient will work as well.

Reply:
Awesome. Thank you

Reply:
This is a critical point and the rule we have for third party providers: we never depend on the official SDK if we can rely on the HttpClient component (and if we must depend on the official SDK, then it's not for Symfony core).

Reply:
Thanks for the clarification. Will update soon 👍

Reply:
When it comes to using HttpClient, don't forge that the official SDK use shortcuts and hooks during authentication process (mostly when it calls the API endpoints), IMO it could be a good idea to use a scoped client 🙂